src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -571,11 +570,13 @@ protected internal override Task OnCloseAsync(TimeSpan timeout)
         protected override void OnClosed()
         {
             base.OnClosed();
-            var httpClientToDispose = _httpClient;
-            if (httpClientToDispose != null)
+            if (_httpClientCache != null && !_httpClientCache.IsDisposed)
             {
-                _httpClient = null;
-                httpClientToDispose.Dispose();
+                lock (_httpClientCache)
```

Comment:
As I detailed in an earlier comment, OnClosed can't be called by multiple threads due to protections in CommunicationObject. If another request is started on another thread, it will add the request to a list of outstanding requests before _httpClientCache would be created. The Close call will block waiting for that outstanding request to complete and then Dispose() after that. The factory state is checked before adding the request to the outstanding requests list so a race condition shouldn't be possible.